balloon-hash/src/error.rs

```diff
@@ -21,6 +21,8 @@ pub enum Error {
     ThreadsTooMany,
     /// Time cost is too small.
     TimeTooSmall,
+    /// Output size not correct.
+    OutputSize,
```

Comment:
Note that adding this error variant makes this PR a breaking one.

Reply:
Yeah, I don't really mind if this can't be merged until the next release. I'm not really sure what else to do here, alternatively of course I can return a wrong error type, just to make it a non-breaking change.

Reply:
This crate doesn’t have many users so you could just go ahead and make a breaking change.

If you also add `#[non_exhaustive]` to the error enum you can make such changes non-breaking going forward.

Reply:
Done.